spyderlib/plugins/editor.py

```diff
@@ -944,7 +952,7 @@ def get_plugin_actions(self):
         self.recent_file_menu = QMenu(_("Open &recent"), self)
         self.recent_file_menu.aboutToShow.connect(self.update_recent_file_menu)
 

```

Comment:
@thewhitetulip, the logic is this: opening a new file and opening the last closed one are different operations, so they don't belong in the same menu.

You can see that by the way this is handled in Firefox. Its File menu has actions to open a new tab and a new window, but not the last closed tab. That option is only present in its Tab context menu (i.e. when you do a right-click over a tab).

Following that logic, that's why I prefer to move this new action to the Editor context menu, which is defined here:

https://github.com/spyder-ide/spyder/blob/master/spyderlib/widgets/sourcecode/codeeditor.py#L2419

If you prefer, we can do that for you after we merge this PR :-)